slides.Rmd

```diff
@@ -547,46 +590,80 @@ baby6 <- read_csv("data/Baby6.csv")
 baby_joined <- baby5 %>%
   left_join(baby6, by = c("FAMILYID", "DOB"))
 ```
-  
 
 
-## Data Visulisation 
+## END BLOCK 3
+
+
+## END BLOCK 4
+
+
+## Data Visulisation - ggplot2
+
+```{r, echo = FALSE}
+# TRAINERS NOTES
+# Maybe we need a slide here giving a bit of background to ggplot2? 
+```
 
 A basic line graph can be created using the following code:
 
 ```{r,  eval=FALSE,echo=TRUE}
+install.packages("ggplot2")
+library(ggplot2)
+
+#Define data
 x <- c(1, 3, 2, 4)
-# Plot the data
-plot(x, type = "l")
-```
+y <- c(4, 8, 7, 11)
 
- Note that `type = "l"` simply means to plot a line.
+# Create a data frame
+df <- as.data.frame(cbind(x, y))
 
+# Plot the data
+line_plot <- ggplot(df, aes(x, y)) + 
+  geom_line() +
+  geom_point()
 
-```{r, echo=FALSE}
-x <- c(1, 3, 2, 4)
-par(mar = c(10, 4, 4, 2))
-plot(x, type = "l")
+# View plot
+line_plot
 ```
 
 
-## Example 
-
+## Example - Barplot
 
 - Read in the `Borders (inc Age).csv` data
 
 ```{r,  eval=FALSE,echo=TRUE}
-borders_age <- read_csv("data/Borders (inc Age).csv")
+borders_age <- read_csv("data/BORDERS (inc Age).csv")
 ```
 
 - Calculate frequencies of `admissionday`
 
 ```{r,  eval=FALSE,echo=TRUE}
-counts <- table(borders_age$admissionday)
-barplot(counts)
+counts <- as.data.frame(table(borders_age$admissionday))
+colnames(counts) <- c("day", "count")
+
+# plot
+bar_plot <- ggplot(counts, aes(x = day, y = count)) +
+  geom_bar(stat = "identity")
```

Comment:
I think `geom_col` is equivalent to `geom_bar(stat = "identity")` and simpler to read (I could be wrong!)

Reply:
I think you are right! `geom_bar()` can make plots of counts directly from the data as well - so you could also do: `ggplot(borders_age) + geom_bar(aes(x = admissionday))` without making a `counts` object.